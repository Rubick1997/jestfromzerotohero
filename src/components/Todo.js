import React from "react";

function Todo({ todo }) {
	const { id, title, completed, description } = todo;

	const h1 = <h1>{title}</h1>;
	const text = completed ? <strike>{h1}</strike> : h1;

	return (
		<div data-testid={`todo-${id}`}>
			{text}-{description}
		</div>
	);
}

export default Todo;
