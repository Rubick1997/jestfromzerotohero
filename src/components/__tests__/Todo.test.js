import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Todo from "../Todo";

test("should render completed todo component", () => {
	const todo = {
		title: "Yoshi's birthday bash",
		details:
			"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
		category: "reminders",
		completed: true,
		id: 1,
	};

	render(<Todo todo={todo} />);
	const todoEl = screen.getByTestId("todo-1");
	expect(todoEl).toBeInTheDocument();
	expect(todoEl).toHaveTextContent("Yoshi's birthday bash");
    expect(todoEl).toContainHTML("<strike><h1>Yoshi's birthday bash</h1></strike>");
});

test("should render not completed todo component", () => {
	const todo = {
		title: "Complete my ninja training",
		details:
			"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took.",
		category: "work",
		completed: false,
		id: 2,
	};

	render(<Todo todo={todo} />);
	const todoEl = screen.getByTestId("todo-2");
	expect(todoEl).toBeInTheDocument();
	expect(todoEl).toHaveTextContent("Complete my ninja training");
    expect(todoEl).not.toContainHTML(
			"<strike><h1>Complete my ninja training</h1></strike>"
		);
});

test("matches snapshot", () => {
  const todo = {
		title: "Complete my ninja training",
		details:
			"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took.",
		category: "work",
		completed: false,
		id: 2,
	};
    const tree = TestRenderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot()
})