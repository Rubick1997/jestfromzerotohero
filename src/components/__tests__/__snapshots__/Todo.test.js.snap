// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-2"
>
  <h1>
    Complete my ninja training
  </h1>
  -
</div>
`;
